scripts/kickstart_generator.rb

```diff
@@ -23,17 +23,17 @@ def initialize(build_base, targets, puddle, manageiq_checkout, appliance_checkou
       @ssui_checkout      = ssui_checkout
     end
 
-    def run(task = :all)
+    def run
+      FileUtils.mkdir_p(@ks_gen_base)
+
       targets.each do |target|
         @target = target # used during ERB evaluation
 
         result = evaluate_erb
 
-        FileUtils.mkdir_p(@ks_gen_base)
-        File.write(@ks_gen_base.join("base.ks"), result)
-
-        write_config(result) if [:all, :config].include?(task)
-        write_json(result)   if [:all, :json].include?(task)
+        file = @ks_gen_base.join("base-#{@target}.ks")
+        $log.info("Writing kickstart file: #{file}") if $log
+        File.write(file, result)
```

Comment:
Previously imagefactory only had a single `--parameters` option, which read a single JSON file containing options. This means that if you had multiple options, you had to stuff them all in there. The result was that the kickstart generator had to write the kickstart file as well as non-kickstart related options in there.

Newer versions have added multiple `--parameter` and `--file-parameter` options (still only a single `--parameter*s*` option though). By using this we can implement a separation of concerns where the kickstart generator only outputs the kickstart file, and we can pass in other options via additional command line arguments.

I could do this in separate PRs but given that I'm already restructuring  part of `vmbuild.rb` I'd prefer not to have to do that extra work. Note that the Vagrant support is a separate commit in the same PR. So it's already somewhat separated.

I've tested the code and all formats build correctly. So I'm pretty sure that we don't need the "Kickstart in JSON" workaround anymore.